packages/core/src/editor/BlockNoteExtensions.ts

```diff
@@ -251,28 +252,112 @@ const getTipTapExtensions = <
       })
     );
     if (opts.collaboration.provider?.awareness) {
-      const defaultRender = (user: { color: string; name: string }) => {
-        const cursor = document.createElement("span");
+      const cursors = new Map<
+        number,
+        { element: HTMLElement; hideTimeout: NodeJS.Timeout | undefined }
+      >();
+
+      const awareness = opts.collaboration!.provider.awareness as Awareness;
```

Comment:
this can be a bit cleaner by

doing `const awareness = opts.collaboration.provider?.awareness` before line 254 and then reusing it (nitpicking a bit)